c2corg_api/views/document.py

```diff
@@ -682,7 +685,17 @@ def get_all_fields(fields, activities, field_list_type):
 def add_load_for_profiles(document_query, clazz):
     if clazz == UserProfile:
         # for profiles load username/name together from the associated user
-        document_query = document_query.options(joinedload('user'))
+        document_query = add_profile_filter(document_query, clazz). \
+            options(contains_eager('user'))
+    return document_query
+
+
+def add_profile_filter(document_query, clazz):
+    if clazz == UserProfile:
+        # make sure only confirmed accounts are returned
+        document_query = document_query. \
+            join(User). \
+            filter(User.email_validated)
```

Comment:
Yes it looks like a test for the existence of any value (is not null). It would be more clear to explicitely check for True.